lib/es-serializer.js

```diff
@@ -350,6 +331,28 @@ ResourceSerializer.sortableShelfMark = (shelfMark) => {
  */
 ResourceSerializer.zeroPadString = (s, padLen = 6) => (new Array(Math.max(0, (padLen - s.length) + 1))).join('0') + s
 
+/**
+ * Returns an object with default shelfMark if shelfMark is missing,
+ * and the sortableShelfMark property, normalized to put items without
+ * shelfMark last
+ */
+ResourceSerializer.normalizedShelfMark = (item, defaultShelfMark) => {
+  // Only set default shelfMark if not electronic
+  if (!item.shelfMark && !item.electronicLocator) {
+    return { shelfMark: defaultShelfMark, sortableShelfMark: 'a' + defaultShelfMark }
```

Comment:
The `sortableShelfMark` value here doesn't look like it will be zero-paded. I think you meant to set it to `sortableShelfMark: 'a' + ResourceSerializer.sortableShelfMark(item['shelfMark'][0])` ?

Reply:
This should be fixed now that we are doing two separate passes to add the default shelf mark and then the `sortableShelfMark`